src/dafny/Relations.dfy

```diff
@@ -26,6 +26,10 @@ module {:options "-functionSyntax:4"} Dafny.Relations {
     forall x, y :: R(x, y) && R(y, x) ==> x == y
   }
 
+  ghost predicate Symmetric<T(!new)>(R: (T, T) -> bool) {
```

Comment:
What's the reason a predicate like this shouldn't be used for heap based types? (`!new`)

Reply:
The reason is quantification: removing `!new` leads to the error: _a forall expression involved in a predicate definition is not allowed to depend on the set of allocated references, but values of 'x' (of type 'T') may contain references (perhaps declare its type as 'T(!new)') (see documentation for 'older' parameters)_.